Update Rails and related gems

awead opened the pull request:
Updates us to the latest compatible version of Rails, which is 6.1.

awead committed:
Update Devise

The latest Devise version, 4.8.0, supports Omniauth 2.0, so we can
remove the custom commit we'd been using.

awead committed:
Update Rails to 6.1.3.2

This includes updates to RSpec, QA, and PaperTrail.

nicktiberi commented on Gemfile:
```diff
@@ -80,7 +78,6 @@ group :test do
   gem 'rspec'
   gem 'rspec-its'
   gem 'rspec-rails', '~> 4.0.0.beta3'
-  gem 'scholarsphere-client'
```
Is `scholarsphere-client` now deprecated in favor of the API methods now built into Scholarsphere?

Did that need to be done in order to upgrade Rails to 6.1? I guess I'm not sure why the file upload process was changed in an upgrade Rails commit.

awead replied:
@nicktiberi I should have put this is a separate commit... It's tangentially related. I'll push up a new commit.

awead replied:
@nicktiberi I pushed up a different commit, and combined one of the previous ones. The `marcel` gem went to 1.0, which blocked the Rails update, but also required a new release of scholarsphere-client. Scholarsphere was using an older version. Updating the client gem caused the one ingest controller test to break because we were using the presigned url business to upload files. Prior to this, we were uploading files to S3 basically by sharing all of the access keys, which isn't a good idea security-wise.

Anyway, the only reason we were using the gem in this case was to craft a POST request to the controller for testing. It didn't seem worthy to go through all the trouble of re-hashing uploads with the new client gem version  just to test this one controller. So I decided to just remove the gem.

awead committed:
Remove scholarsphere-client

Changes to the client gem made testing more difficult. Since we were
only using it to test the ingest controller, it makes more sense to take
a different approach and use the internal S3 client to set up the test
instead of the client gem. This was a case where it was causing more
problems than it solved. The integration with the client and
Scholarsphere is still being tested in the client itself.

nicktiberi reviewed (approved):
Looks good! 🎉
